keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepositoryTests.kt

```diff
@@ -294,5 +306,36 @@ abstract class CombinedRepositoryTests<D : DeliveryConfigRepository, R : Resourc
 
     context("persisting delivery config manifests") {
     }
+
+    context("don't allow resources to be managed by more than 1 config") {
+      before {
+        subject.upsertDeliveryConfig(deliveryConfig)
+      }
+      test("trying to persist another config with the same resource") {
+        expectThrows<DuplicateManagedResourceException> { subject.upsertDeliveryConfig(secondDeliveryConfig) }
+        expectThrows<NoSuchDeliveryConfigException> { deliveryConfigRepository.get(secondConfigName) }
+        expectCatching { resourceRepository.get(firstResource.id) }.isSuccess()
+      }
+
+      context("delete the resource from the first config, and try to persist again") {
```

Comment:
This test probably isn't necessary, since it's just testing normal saving. Do you feel like it adds any other value?

Reply:
not really :) I can remove those tests.

Reply:
I lean towards no, because this is more of a unit test. I think there's a better place to test "if resource doesn't exist it is saved" (in the deliveryConfigRepositoryTests, already exists). I think it's really important to test that we error on a bad condition here, but not the remediation steps because that's just the happy path.

Reply:
ok, that makes sense to me. will remove it